### PR TITLE
Sign APKs with apksigner instead of via Gradle

### LINKED ISSUE #414: F-Droid reproducible build failed
**Describe the bug**
```
DOES NOT VERIFY
ERROR: APK Signature Scheme v2 signer #1: APK integrity check failed. CHUNKED_SHA256 digest mismatch. Expected: <a4192e5d7a62b86c4c02a19860b9984102da180d9a6e0ad9707029beb10e91c0>, actual: <11e234eb288257ac5c47e1090a760391291155ea0d4b59e8e82398a8908e0c47>
```

Colud you please take a look? Thanks!

### PR DESCRIPTION
**Pull request by h1dden-da3m0n**
### Description

This PR aims to fix our issues with the F-Droid build and our APKs signed via Gradle and not `apksigner` by reverting to using `apksigner`.
Big THANK YOU to @CarlosOlivo figuring this one out!

### Changes

* change apk signing from gradle to apksigner

### Issues

* should fix #414

### CODE REVIEW

**Review by Maxr1998 — APPROVED** [review]

File: .github/workflows/app-publish.yaml [hunk 1]
```diff
@@ -20,17 +20,30 @@ jobs:
...
+        uses: r0adkll/sign-android-release@v1
+        with:
+          releaseDirectory: app/build/outputs/apk/libre/release
+          signingKeyBase64: ${{ secrets.KEYSTORE }}
+          keyStorePassword: ${{ secrets.KEYSTORE_PASSWORD }}
+          alias: ${{ secrets.KEY_ALIAS }}
+          keyPassword: ${{ secrets.KEY_PASSWORD }}
+      - name: Sign Proprietary APK
+        uses: r0adkll/sign-android-release@v1
+        id: proprietarySign
```
Comment (nielsvanvelzen) [inline .github/workflows/app-publish.yaml:35]:
```suggestion
        id: proprietarySign
        uses: r0adkll/sign-android-release@v1
```

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: .github/workflows/app-publish.yaml (hunk 1 above)
Comment (h1dden-da3m0n) [reply .github/workflows/app-publish.yaml:35]: fine lets 'fix'/unify the ordering xD

File: .github/workflows/app-publish.yaml [hunk 2]
```diff
@@ -20,17 +20,30 @@ jobs:
       - name: Set JELLYFIN_VERSION
         run: echo "JELLYFIN_VERSION=$(echo ${GITHUB_REF#refs/tags/v} | tr / -)" >> $GITHUB_ENV
       - name: Assemble release APKs
-        env:
-          KEYSTORE: ${{ secrets.KEYSTORE }}
-          KEYSTORE_PASSWORD: ${{ secrets.KEYSTORE_PASSWORD }}
-          KEY_ALIAS: ${{ secrets.KEY_ALIAS }}
-          KEY_PASSWORD: ${{ secrets.KEY_PASSWORD }}
         run: ./gradlew --no-daemon --info assembleRelease versionTxt
+      - name: Sign Libre APK
```
Comment (nielsvanvelzen) [inline .github/workflows/app-publish.yaml:24]:
```suggestion
      - name: Sign libre APK
```

File: .github/workflows/app-publish.yaml [hunk 3]
```diff
@@ -20,17 +20,30 @@ jobs:
...
+        id: libreSign
+        uses: r0adkll/sign-android-release@v1
+        with:
+          releaseDirectory: app/build/outputs/apk/libre/release
+          signingKeyBase64: ${{ secrets.KEYSTORE }}
+          keyStorePassword: ${{ secrets.KEYSTORE_PASSWORD }}
+          alias: ${{ secrets.KEY_ALIAS }}
+          keyPassword: ${{ secrets.KEY_PASSWORD }}
+      - name: Sign Proprietary APK
```
Comment (nielsvanvelzen) [inline .github/workflows/app-publish.yaml:33]:
```suggestion
      - name: Sign proprietary APK
```

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

**Review by nielsvanvelzen — APPROVED** [review]

**Review by Maxr1998 — APPROVED** [review]

**Pull request merged**